sdk/search/azure-search-documents/azure/search/documents/_service/_datasources_client.py

```diff
@@ -80,20 +81,26 @@ def create_datasource(self, data_source, **kwargs):
     def create_or_update_datasource(self, data_source, name=None, **kwargs):
         # type: (DataSource, Optional[str], **Any) -> Dict[str, Any]
         """Creates a new datasource or updates a datasource if it already exists.
-
         :param name: The name of the datasource to create or update.
         :type name: str
         :param data_source: The definition of the datasource to create or update.
         :type data_source: ~search.models.DataSource
+        :keyword only_if_unchanged: If set to true, the operation is performed only if the
+        e_tag on the server matches the e_tag value of the passed data_source.
+        :type only_if_unchanged: bool
```

Comment:
Why not use https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/core/azure-core#matchconditions?

Reply:
It’s by choice. Keeping it consistent with other languages by using a single Boolean instead of match conditions. See  issue description for more details.

Reply:
No other library exposes matchconditions. In fact, the guidelines actually suggest using `onlyIfUnchanged` (and similar words in other cases). @annelo-msft for AppConfiguration formatlized much of this.

Reply:
Python is using MatchConditions. We are not using the boolean flag.

Reply:
There is no scenario for IfNoneMatch either. This doesn't seem like an idiomatic difference, so why not be consistent across the languages given the [guidelines](https://azure.github.io/azure-sdk/general_design.html#conditional-requests)?

Reply:
We already have precedence for this in Python. Language consistency trumps cross language consistency. Python should use MatchConditions.